File: src/main/java/com/gitlab/daring/image/command/feature/DetectFeaturesCommand.java
package com.gitlab.daring.image.command.feature;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import org.bytedeco.javacpp.opencv_xfeatures2d.SURF;

public class DetectFeaturesCommand extends BaseCommand {

    public DetectFeaturesCommand(String... args) {
        super(args);
    }

    @Override
    public void execute(CommandEnv env) {
        if (env.featureDetector == null) env.featureDetector = SURF.create();
        env.featureDetector.detect(env.mat, env.keyPoints);
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/feature/DetectFeaturesCommand.kt
package com.gitlab.daring.image.command.feature

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand

internal class DetectFeaturesCommand(args: Array<String>) : KBaseCommand(args) {

    override fun execute(env: CommandEnv) {
        env.featureDetector.detect(env.mat, env.keyPoints)
    }

}